core/src/main/java/uk/gov/justice/services/core/extension/AnnotationScanner.java

```diff
@@ -41,15 +43,21 @@
 
     @SuppressWarnings("unused")
     void afterDeploymentValidation(@Observes final AfterDeploymentValidation event, final BeanManager beanManager) {
-        beanManager.getBeans(Object.class, annotationLiteral()).stream()
+        final Set<Bean<?>> beans = beanManager.getBeans(Object.class, annotationLiteral());
```

Comment:
Rather than Object.class can you just get those for your Interceptor.class interface? - shortcutting the process, or maybe something similar to this, see my comment on the new annotation.

Also I think it was suggested that perhaps if we were to start adding even more annotations we should refactor this code to make it cleaner rather than running through the streams multiple times.